include/seqan3/core/algorithm/concept.hpp

```diff
@@ -32,6 +32,18 @@ struct pipeable_config_element;
 namespace seqan3::detail
 {
 
+// ----------------------------------------------------------------------------
+// compatibility_table
+// ----------------------------------------------------------------------------
+
+/*!\brief Declaration of algorithm specific compatibility table.
+ * \ingroup algorithm
+ * \tparam algorithm_id_type The type of the algorithm specific id. Algorithm configurations must maintain
+ *                           this table to allow validation checks.
+ */
+template <typename algorithm_id_type>
+inline constexpr std::array<std::array<void, 0>, 0> compatibility_table;
```

Comment:
When working with clang it showed that `void` is an invalid type for array, because `reference` is defined as `value_type&` and that is an invalid type.

But we can fix that in a follow up PR.